pints/_mcmc/_dram_ac.py

```diff
@@ -84,131 +62,91 @@ class DramACMC(pints.AdaptiveCovarianceMC):
         return min(0, alpha_log)
 
-    def _calculate_r_log(self, fx):
+    def _alpha_2_log(self, fx, fy1, fy2):
         """
-        Calculates value of logged acceptance ratio (eq. 3 in [1]_).
+        Calculates probability of acceptance in stage 1 of DRAM (eq. 2 in
+        [1]_).
         """
-        c = self._proposal_count
-        temp_Y = np.concatenate([[self._current], self._Y[0:(c + 1)]])
-        temp_log_Y = np.concatenate(
-            [[self._current_log_pdf], self._Y_log_pdf[0:(c + 1)]])
-        self._r_log = self._calculate_alpha_log(c, temp_Y, temp_log_Y)
+        alpha_log = (fy2 - fx +
+                     np.log1p(np.exp(self._alpha_1_log(fy2, fy1))) -
+                     np.log1p(np.exp(self._alpha_1_log(fx, fy1))))
+        return min(0, alpha_log)
 
     def _generate_proposal(self):
         """ See :meth:`AdaptiveCovarianceMC._generate_proposal()`. """
-        if self._before_kernels_set:
-            self.set_sigma_scale()
-            self._Y = [None] * self._n_kernels
-            self._Y_log_pdf = np.zeros(self._n_kernels)
-            self._before_kernels_set = False
-
         proposed = np.random.multivariate_normal(
             self._current, np.exp(self._log_lambda) *
             self._sigma[self._proposal_count])
-        self._Y[self._proposal_count] = proposed
         return proposed
 
     def name(self):
         """ See :meth:`pints.MCMCSampler.name()`. """
         return 'Delayed Rejection Adaptive Metropolis (Dram) MCMC'
 
-    def n_kernels(self):
-        """ Returns number of proposal kernels. """
-        return self._n_kernels
-
     def n_hyper_parameters(self):
         """ See :meth:`TunableMethod.n_hyper_parameters()`. """
-        return 3
+        return 2
+
+    def _r_log(self):
+        """
+        Calculates value of logged acceptance ratio (eq. 3 in [1]_).
+        """
+        if self._proposal_count == 0:
+            r_log = self._alpha_1_log(
+                self._current_log_pdf, self._Y_log_pdf[0])
+        else:
+            r_log = self._alpha_2_log(
+                self._current_log_pdf, self._Y_log_pdf[0],
+                self._Y_log_pdf[1]
+            )
+        return r_log
 
     def set_hyper_parameters(self, x):
         """
-        The hyper-parameter vector is ``[eta, n_kernels, upper_scale]``.
+        The hyper-parameter vector is ``[eta, sigma_scale]``.
 
         See :meth:`TunableMethod.set_hyper_parameters()`.
         """
         self.set_eta(x[0])
-        self.set_n_kernels(x[1])
-        self.set_upper_scale(x[2])
-
-    def set_n_kernels(self, n_kernels):
-        """ Sets number of proposal kernels. """
-        if n_kernels < 1:
-            raise ValueError('Number of proposal kernels must be equal to ' +
-                             'or greater than 1.')
-        self._n_kernels = int(n_kernels)
+        self.set_sigma_scale(x[1])
 
-    def set_upper_scale(self, upper_scale):
+    def set_sigma_scale(self, scales):
```

Comment:
Will users want to put in a list here, or should we just make the arguments be scale1 and scale2?
(Question, not a suggestion! I have no idea)

Reply:
Actually, change this to scale1 and scale2 please, so that you resolve the issue with hyperparameters too